src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +114,660 @@ const ActorInit En_Nb_InitVars = {
+            }
+        }
+    }
+
+    if (this->stateFlags & EN_NB_FLAG_100) {
+        this->stateFlags &= ~EN_NB_FLAG_100;
+        this->headRotZ = 0;
+        this->headRotY = 0;
+        this->unk_282 = 20;
+    } else if (DECR(this->unk_282) == 0) {
+        this->stateFlags |= EN_NB_FLAG_400;
+    }
+}
+
+// Related to both stories?
+void func_80BC08E0(EnNb* this, PlayState* play) {
+    if (this->unk_284 == 0) {
+        EnNb_ChangeAnim(this, EN_NB_ANIM_TALK_ONCE);
+        this->stateFlags |= EN_NB_FLAG_400;
+        this->unk_284++;
+    } else if ((this->unk_284 == 1) && Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        EnNb_ChangeAnim(this, EN_NB_ANIM_1);
+        this->stateFlags &= ~EN_NB_FLAG_400;
+        this->unk_284++;
+    }
+}
+
+void func_80BC0978(EnNb* this, PlayState* play) {
+    if (this->unk_284 == 0) {
+        EnNb_ChangeAnim(this, EN_NB_ANIM_RELIEVED);
+        this->stateFlags &= ~EN_NB_FLAG_20;
+        this->stateFlags |= EN_NB_FLAG_400;
+        this->unk_284++;
+    } else if ((this->unk_284 == 1) && Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        EnNb_ChangeAnim(this, EN_NB_ANIM_TALK_LOOP);
+        this->stateFlags &= ~EN_NB_FLAG_400;
+        this->unk_284++;
+    }
+}
+
+s32 func_80BC0A18(EnNb* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    u16 currentTextId = play->msgCtx.currentTextId;
+
+    if (player->stateFlags1 & PLAYER_STATE1_40) {
+        this->stateFlags |= EN_NB_FLAG_80;
+
+        if (this->textId != currentTextId) {
+            switch (currentTextId) {
+                case 0x28CF:
+                    this->stateFlags |= EN_NB_FLAG_20;
+                    EnNb_ChangeAnim(this, EN_NB_ANIM_TALK_LOOP);
+                    break;
+
+                case 0x2904: // "You want to hear the carnival of time story? ..."
+                case 0x290B: // "You want to hear the four giants story? ..."
+                    this->unk_18C = func_80BC08E0;
+                    this->unk_284 = 0;
+                    break;
+
+                case 0x28CD:
+                    this->unk_18C = func_80BC0978;
+                    this->unk_284 = 0;
+                    break;
+
+                case 0x28CB: // "I told you I already ate!"
+                    EnNb_ChangeAnim(this, EN_NB_ANIM_ANGRY);
+                    break;
+
+                case 0x28C7:
+                case 0x2901:
+                case 0x2902:
+                case 0x2906:
+                case 0x290D:
+                case 0x2912:
+                    EnNb_ChangeAnim(this, EN_NB_ANIM_TALK_LOOP);
+                    break;
+            }
+        }
+
+        this->textId = currentTextId;
+    } else if (this->stateFlags & EN_NB_FLAG_80) {
+        this->unk_18C = NULL;
+        this->textId = 0;
+        this->stateFlags &= ~EN_NB_FLAG_80;
+        EnNb_ChangeAnim(this, EN_NB_ANIM_1);
+    }
+
+    if (this->unk_18C != NULL) {
+        this->unk_18C(this, play);
```

Comment:
Given that this is run here and not in Update it's not really an ActionFunc, just happens to share the prototype.